runtime/queries/metricsview_aggregation.go

```diff
@@ -471,6 +471,9 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 			*q.Limit = 100
 		}
 		limitClause = fmt.Sprintf("LIMIT %d", *q.Limit)
+	} else if q.PivotOn != nil {
+		l := 1_000_000 / cols
+		limitClause = fmt.Sprintf("LIMIT %d", l)
 	}
```

Comment:
1. Pull the number into a constant at the top of the file, as suggested in the previous comment
2. We need to return an error if the limit is exceeded (for both DuckDB and Druid). Since otherwise people would get partial/wrong results. So the limit probably needs to be `maxPivotCells + 1` and then after querying, it count the number of rows and check it's `<= maxPivotCells`.